### ISSUE TITLE
equation numbering not working for epub output format

### ISSUE THREAD

**Issue by francoisbirgand** [rstudio/bookdown#530]
Hello,

first I would like to thank you all for this incredible package you have put together. I am a university professor and I believe it is transformative and certainly is a fantastic tool for reproducible research. I have started asking my students to write everything in R markdown, and I would like to ask them to write their thesis in bookdown...?!! I am not a Latex person, I try to be an `R` one.

Now, I have started to use bookdown to write a textbook for my classes. And I have stumbled upon an issue with equations. After many hours spent to find my mistakes, I decided to copy and paste equation 2.2 from the bookdown manual (https://bookdown.org/yihui/bookdown/).

So, for this I copy the exact code below in the `02-literature.Rmd` document and I clicked `Build Book`.

```
# Literature

Here is a review of existing methods.

$$
\begin{align} 
g(X_{n}) &= g(\theta)+g'({\tilde{\theta}})(X_{n}-\theta) \notag \\
\sqrt{n}[g(X_{n})-g(\theta)] &= g'\left({\tilde{\theta}}\right)
  \sqrt{n}[X_{n}-\theta ] (\#eq:align)
\end{align}
$$
```
And I get this message:

```
output file: bookdown-demo.knit.md

! Package amsmath Error: Erroneous nesting of equation structures;
(amsmath)                trying to recover with `aligned'.

Error: Failed to compile bookdown-demo.tex. See bookdown-demo.log for more info.
```
So, as I was writing this I solved one issue and I suggest this becomes written on the bookdown manual.  The manual could be clearer by saying that if one wants to reference equations, DO NOT USE `$$` before or after the equation like I have in the code above.  This could save many hours of frustration to many people. By the way, the code above will work for `gitbook` output but not `pdf_book` output.

Looking back in the manual, the `$$` are not there, but they are on the previous page/part 2.1.3. So, if I may suggest so, adding this warning on top of part 2.2 would be great! 

I did find one issue, which is that the `epub` output will not number the equation using this code:

```# Literature

Here is a review of existing methods.


\begin{align} 
g(X_{n}) &= g(\theta)+g'({\tilde{\theta}})(X_{n}-\theta) \\
\sqrt{n}[g(X_{n})-g(\theta)] &= g'\left({\tilde{\theta}}\right)
  \sqrt{n}[X_{n}-\theta ] (\#eq:align)
\end{align} 
```
it yields:

<img width="554" alt="screen shot 2018-02-05 at 11 27 03 am" src="https://user-images.githubusercontent.com/31140209/35815887-89297234-0a67-11e8-89a0-23c232822ed5.png">

Thanks again for your fantastic contribution to all, and certainly my field!
Best.

**Comment by dataopt** [comment]
I also found out the hard way that `\begin{align}`, being a LaTex environment, should *not* be enclosed with $$.  This is not really a bookdown issue but rather a pandoc issue.

Lots of things don't work in EPUB unfortunately.  I think equation labelling seems to be one of them.

**Comment by yihui** [comment]
Yeah, bookdown primarily supports HTML and PDF output formats. EPUB and Word are hard.

The LaTeX math requires knowledge about LaTeX, and you just write the expressions in the exact way you write in LaTeX. In LaTeX, there should not be `$$` around equation environments.

I understand this could be confusing, but it is how LaTeX works. I think it makes sense to add a reminder in the book (I thought my examples were clear enough but apparently this point is worth a special emphasis).

A pull request will be welcome!

**Comment by antaldaniel** [comment]
I wonder who can you figure out a in a chunk what is the current format, similarly to `knitr::is_latex_output `and `knitr::is_html_output` if epub is being made. Currently I use a workaround to create different tables for latex, html output, and the other option is to use `kableExtra::kable_as_image` for Word. But it may be useful to differentiate between Word and EPUB, what would be the best way to do that?

**Comment by tchevri** [comment]
In my setup chunk, i have:

`
    ````{r setup, echo=FALSE, message=FALSE, include=FALSE, warning=FALSE}  
    ## CHECK COMPILATION TYPE  
    out_type <- knitr::opts_knit$get("rmarkdown.pandoc.to")  
    ````
`

Then in your code, all you need is

`
    ```{r chunk_name, eval = (out_type=="epub3"), echo=FALSE, results='asis'}  
    ```
`

for instance.

**Comment by cderv** [comment]
Just a not for later reference if we tackle this issue: 
Math in EPUB is not something that will always renders nicely: https://pandoc.org/epub.html#math

One solution that helps is using `--webtex` math mode in Pandoc to render math as an image in the document. This does not support numbering but we can try to add the necessary missing piece for numbering and referencing as we are doing for figures. Not sure what it would cost to support this but seems possible.
